src/python/pants/testutil/rule_runner.py

```diff
@@ -310,17 +310,21 @@ def rewrite_rule_for_inherent_environment(rule):
         self.build_config = build_config_builder.create()
 
         self.environment = CompleteEnvironmentVars({})
-        self.options_bootstrapper = self.create_options_bootstrapper(args=bootstrap_args, env=None)
         self.extra_session_values = extra_session_values or {}
         self.inherent_environment = inherent_environment
         self.max_workunit_verbosity = max_workunit_verbosity
-        options = self.options_bootstrapper.full_options(
-            self.build_config,
-            union_membership=UnionMembership.from_rules(
-                rule for rule in self.rules if isinstance(rule, UnionRule)
-            ),
-        )
-        global_options = self.options_bootstrapper.bootstrap_options.for_global_scope()
+
+        # Change cwd and add sentinel file (BUILDROOT) so NativeOptionParser can find build_root.
+        with pushd(self.build_root):
+            Path("BUILDROOT").touch()
```

Comment:
I opted to use `BUILDROOT` instead of `BUILD_ROOT` so that anyone who needs to inspect a sandbox has a better idea where the file comes from (or can more easily grep for where it comes from).

Reply:
I didn't fully understand this comment. I'm fine with either one, but did you mean, using BUILDROOT in lieu of BUILD_ROOT so that it lines up with the other usages?

Reply:
There is a `BUILD_ROOT` file at the root of the pants repo. Some tests depend on that (`//BUILD_ROOT:files`), so it gets copied to the root of the pytest sandbox.

Now, RuleRunner also adds a sentinel file to its own sandboxes. Using the variant with `_` gives a subtle indicator of why the file is in the sandbox / what created the file; It gives a subtle indicator that the file is not a copy of the file that `//:BUILD_ROOT:files` pulls in.